server/functional-test/src/test/java/org/zanata/feature/administration/AutoRoleAssignmentTest.kt

```diff
@@ -18,55 +18,54 @@
  * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
  * site: http://www.fsf.org.
  */
-package org.zanata.feature.administration;
+package org.zanata.feature.administration
 
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.zanata.feature.Trace;
-import org.zanata.feature.testharness.TestPlan.DetailedTest;
-import org.zanata.feature.testharness.ZanataTestCase;
-import org.zanata.page.administration.RoleAssignmentsPage;
-import org.zanata.workflow.LoginWorkFlow;
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.zanata.feature.Trace
+import org.zanata.feature.testharness.TestPlan.DetailedTest
+import org.zanata.feature.testharness.ZanataTestCase
+import org.zanata.workflow.LoginWorkFlow
 
-import static org.assertj.core.api.Assertions.assertThat;
+import org.assertj.core.api.Assertions.assertThat
 
 /**
- * @author Damian Jansen <a
- *         href="mailto:djansen@redhat.com">djansen@redhat.com</a>
+ * @author Damian Jansen [djansen@redhat.com](mailto:djansen@redhat.com)
  */
-@Category(DetailedTest.class)
-public class AutoRoleAssignmentTest extends ZanataTestCase {
+@Category(DetailedTest::class)
+class AutoRoleAssignmentTest : ZanataTestCase() {
 
     @Trace(summary = "The administrator can create a rule to assign roles " +
             "at user sign in")
-    @Test(timeout = ZanataTestCase.MAX_SHORT_TEST_DURATION)
-    public void createAutoRoleAssignment() throws Exception {
-        RoleAssignmentsPage roleAssignmentsPage = new LoginWorkFlow()
+    @Test(timeout = MAX_SHORT_TEST_DURATION.toLong())
+    @Throws(Exception::class)
+    fun createAutoRoleAssignment() {
+        val roleAssignmentsPage = LoginWorkFlow()
                 .signIn("admin", "admin")
                 .goToAdministration()
                 .goToManageRoleAssignments()
                 .clickMoreActions()
                 .clickCreateNew()
                 .enterIdentityPattern(".+ransla.+")
                 .selectRole("admin")
-                .saveRoleAssignment();
+                .saveRoleAssignment()
 
-        assertThat(roleAssignmentsPage.getRulesByPattern())
-                .as("The rule was created")
-                .contains(".+ransla.+");
+        assertThat(roleAssignmentsPage.rulesByPattern)
+                .`as`("The rule was created")
```

Comment:
This can be nicer in Kotlin with the name `describedAs` (no need for back-ticks).